modules/virtual_environments/auto-venv/venv_helpers.nu

```diff
@@ -23,6 +23,20 @@ export def venv-is-active [] {
     '__auto_venv' in (overlay list)
 }
 
+# Creates a virtual environment under the current directory
+export def 'venv-create' [] {
+	let venv_path = $env.PWD
+	let venv_name = ($env.PWD | path basename) 
+
+	let sys_posix = ['darwin', 'linux', 'unix', 'posix', 'gnu']
+	let is_posix = (((sys).host.name | str downcase) in $sys_posix)
+	let python_name = if $is_posix {'python3'} else {'py.exe'}
+	$python_name -m venv .venv --clear --prompt $venv_name
+	$'.venv/bin/($python_name)' -m pip install -U pip wheel setuptools
```

Comment:
I thought you had to have a caret here to call externals, like this
```suggestion
	^$'.venv/bin/($python_name)' -m pip install -U pip wheel setuptools
```

Reply:
ah, ok. ya, `print (blah)` would work.

Reply:
but `^$python_name` shouldn't do harm to the previous line :wink:

Reply:
`^$'.venv/bin/($python_name)' -m pip install -U pip wheel setuptools` will execute this line versus just print it.

Reply:
No, I wanted it to execute `python3 -m ...` but that doesn't actually run it. I could `(exec "python3 -m ...")` but not sure if that is the best way (do not want to evaluate raw strings...)

Is there a better way to call a command name `"python3"` stored in a variable? How does one use the variable `CC` storing `"gcc9"` or `"clang"` in nushell?

Reply:
I believe that you call external commands in variables like `^$"($command_variable)"`. If that doesn't work, it's probably a bug.

Reply:
even simpler,
```bash
let cmd = "nvim"
^$cmd
```
runs the `nvim` command and opens my editor :ok_hand:

> **Note**  
> ```bash
> ^$"($cmd)"
> ```
> and
> ```bash
> run-external $cmd
> ```
> also work